internal/controller/job/jobutil.go

```diff
@@ -20,6 +20,14 @@ import (
 	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// isBackoffLimitExceeded returns true if the jobs backoff limit has been exceeded
+func isBackoffLimitExceeded(job *apiv1.Job) bool {
+	if job.Status.Failed >= *job.Spec.BackoffLimit {
```

Comment:
```go
return job.Status.Failed >= *job.Spec.BackoffLimit
```

Reply:
Added a `nil` check for `BackoffLimit` per @cbandy's comment, and if that passes now returning per your recommendation here.